assets/js/controllers/settings/manageSecondPassword.controller.js

```diff
@@ -61,4 +69,44 @@ function ManageSecondPasswordCtrl ($rootScope, $scope, Wallet, $timeout, MyWalle
     $scope.status.waiting = true;
     Wallet.setSecondPassword($scope.fields.password, success);
   };
+
+  $scope.recoveryModal = () => {
+    const openModal = () => $uibModal.open({
+      templateUrl: 'partials/recovery-before-second-password.jade',
+      controller: 'RecoveryBeforeSecondPasswordCtrl',
+      windowClass: 'bc-modal'
+    });
+
+    if (!Wallet.status.didConfirmRecoveryPhrase && !Wallet.settings.secondPassword) {
+      openModal();
```

Comment:
`$uibModal.open(...).result` is a promise that rejects if the modal was "dismissed" and resolves if the modal was "closed". Using this promise to detect what the user does with the modal would be more clean than having a `dismissedRecoveryPrompt` variable and watching it. Let me know if you want to talk about this, would be happy to help.